presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -653,9 +656,29 @@ private AWSCredentialsProvider getAwsCredentialsProvider(URI uri, Configuration
             return new InstanceProfileCredentialsProvider();
         }
 
+        String customCredentialsProviderClass = conf.get(S3_CREDENTIALS_PROVIDER);
+        if (customCredentialsProviderClass != null && !customCredentialsProviderClass.isEmpty()) {
+            return getCustomAWSCredentialsProvider(uri, conf, customCredentialsProviderClass);
+        }
+
         throw new RuntimeException("S3 credentials not configured");
     }
 
+    private AWSCredentialsProvider getCustomAWSCredentialsProvider(URI uri, Configuration conf, String customCredentialsProviderClass)
+    {
+        try {
+            AWSCredentialsProvider credentials = (AWSCredentialsProvider) Class
+                    .forName(customCredentialsProviderClass)
```

Comment:
In general, I think we want to follow what the other Hadoop code does, and everything I know of uses the conf class to do loads.  Specifically, it is important to use the classloader of the plugin load these classes since Presto uses separate class loaders for the core engine and plugins.